src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -1064,15 +1063,8 @@ impl_commit_tree (RpmOstreeTreeComposeContext *self,
 
   if (self->treefile_rs)
     {
-      if (!rpmostree_check_passwd (self->repo, self->rootfs_dfd, **self->treefile_rs,
-                                   self->treefile, self->previous_checksum,
-                                   cancellable, error))
-        return glnx_prefix_error (error, "Handling passwd db");
-
-      if (!rpmostree_check_groups (self->repo, self->rootfs_dfd, **self->treefile_rs,
-                                   self->treefile, self->previous_checksum,
-                                   cancellable, error))
-        return glnx_prefix_error (error, "Handling group db");
+      rpmostreecxx::check_passwd_group_entries (*self->repo, self->rootfs_dfd,
+                                                **self->treefile_rs, self->previous_checksum);
```

Comment:
Self-note: a NULL `previous_checksum` will panic here, the argument needs to be conditionally set empty instead.